paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java

```diff
@@ -99,8 +99,13 @@ public static <KEY> DataStream<RowData> sortStreamByKey(
                     "The adaptive batch scheduler is not supported. Please set the sink parallelism using the key: "
                             + FlinkConnectorOptions.SINK_PARALLELISM.key());
         }
-        final int sampleSize = sinkParallelism * 1000;
-        final int rangeNum = sinkParallelism * 10;
+        final int localSampleSize =
+                options.getLocalSampleSize().orElseGet(() -> sinkParallelism * 1000);
+        final int globalSampleSize =
+                options.getGlobalSampleSize().orElseGet(() -> sinkParallelism * 1000);
+        final int rangeNum = options.getSortRange().orElseGet(() -> sinkParallelism * 10);
```

Comment:
I think there is no need to make localSampleSize and globalSampleSize different, if you want a performance improvement when parallelism is large, just add one parameter of sampleSize option. 20 or 2000 times, default 1000. rangeNum still sinkParallelism * 10.